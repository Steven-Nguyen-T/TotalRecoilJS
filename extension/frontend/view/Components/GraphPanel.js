import React, { useEffect, useState } from 'react';
import {
  BrowserRouter as Router,
  Switch,
  Route,
  Link,
  Redirect
} from "react-router-dom";

import SelectorTree from '../SelectorTree'
import TreeView from '../TreeView';


const GraphPanel = (props) => {
  const [view, setView] = useState(true);
  const [viewName, setviewName] = useState('Component Tree')

  const changeView = () => {
    setView(!view);
    // change text of the button
    // buttonName === 'Component Tree' ? 
    if (viewName === 'Component Tree') {
      setviewName('Selector Tree')
    } else {
      setviewName('Component Tree')
    }

    // console.log(view);
  }

  return (
    <div className='graph-container'>
      <h1 className='h1-absolute' >{viewName}</h1>
      <button onClick={changeView} className='graph-button-1'>Change View</button>
      {view ? <TreeView tree={props.tree} /> : <SelectorTree tree={props.tree} />}
    </div>

    // <Router>
    //   <div className='graphDisplays'>
    //     <div id='graph-options'>
    //       <Link to="/"><button onClick={changeView} className='graph-button-1'>Component Tree</button></Link>
    //       <Link to="/stateTree"><button className='graph-button-2'>Selectors and Atoms</button></Link>
    //     </div>
    //     <Switch>
    //       <Route
    //         exact
    //         path="/"
    //         render={() => {
    //           console.log(view)
    //           return (
    //             view ? <Redirect to="/" /> : <Redirect to="/stateTree" />
    //           )
    //         }
    //         }
    //       >
    //       </Route>

    //       <Route path="/stateTree"><SelectorTree tree={props.tree} /></Route>
    //       <Route path="/"><TreeView tree={props.tree} /></Route>
    //     </Switch>
    //   </div>
    // </Router>
  )
};

export default GraphPanel;