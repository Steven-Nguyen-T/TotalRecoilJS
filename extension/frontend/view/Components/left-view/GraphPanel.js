import React, { useEffect, useState } from 'react';
import {
  BrowserRouter as Router,
  Switch,
  Route,
  Link,
  Redirect
} from "react-router-dom";

import SelectorTree from './SelectorTree'
import TreeView from './treeView';


const GraphPanel = (props) => {
  const [view, setView] = useState(true);
  const [viewName, setviewName] = useState('Component Tree')

  const changeView = () => {
    setView(!view);

    if (viewName === 'Component Tree') {
      setviewName('Selector Tree')
    } else {
      setviewName('Component Tree')
    }
  }

  return (
    <div className='graph-container'>
      <h1 className='h1-absolute' >{viewName}</h1>
      <button onClick={changeView} className='graph-button-1'>Change View</button>
      {view ? <TreeView tree={props.tree} /> : <SelectorTree tree={props.tree} />}
    </div>
  )
};

export default GraphPanel;