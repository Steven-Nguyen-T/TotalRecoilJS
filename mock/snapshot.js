export const tree = [
  {
    children: [{
      name: 'test Recoil Root',
      children: [
        {name: 'Body'},
        {name: 'List', atoms: ['atom1', 'atom2']}
      ]  
    }], 
    name: 'test App'
  },
  {
    atomVal: {atom1: 'testing123', atom2: 'testing098'}
  },
  {
    name: 'Selector tree test',
    children: [
      {
        name: 'selector1', children: [{name: 'atom1', value: 100}]
      }, 
      {
        name: 'selector2', children: [{name: 'atom2', value: 100}]
      }
    ]
  }
]