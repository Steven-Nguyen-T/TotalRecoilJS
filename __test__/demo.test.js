import React from 'react';
import GraphPanel from '../extension/frontend/view/Components/GraphPanel';
import { tree } from '../mock/snapshot.js';
import {
  render,
  cleanup,
} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import '@babel/polyfill';

afterEach(cleanup);

it('renders & matches snapshot - no props', () => {
  const {asFragment} = render(<GraphPanel />);
  expect(asFragment()).toMatchSnapshot();
});

it('renders & matches snapshot - Tree props', () => {
  const {asFragment} = render(
    <GraphPanel
      tree={ tree }
    />,
  );
  expect(asFragment()).toMatchSnapshot();
});