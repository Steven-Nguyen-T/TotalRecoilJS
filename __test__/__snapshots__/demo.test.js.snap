// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders & matches snapshot - componetAtomTree props 1`] = `
<DocumentFragment>
  <div
    class="graph-container"
  >
    <h1
      class="h1-absolute"
    >
      Component Tree
    </h1>
    <button
      class="graph-button-1"
    >
      Change View
    </button>
    <div
      id="canvas"
    >
      <svg
        height="768"
        width="512"
      >
        <g
          transform="translate(100, -60)"
        >
          <g
            class="node"
            transform="translate(0,256)"
          >
            <circle
              fill="steelblue"
              r="6"
            />
            <text
              dy="0.31em"
              text-anchor="start"
              x="6"
            >
              test App
            </text>
          </g>
          <g
            class="node"
            transform="translate(95,256)"
          >
            <circle
              fill="steelblue"
              r="6"
            />
            <text
              dy="0.31em"
              text-anchor="start"
              x="6"
            >
              test Recoil Root
            </text>
          </g>
          <g
            class="node"
            transform="translate(190,128)"
          >
            <circle
              fill="steelblue"
              r="6"
            />
            <text
              dy="0.31em"
              text-anchor="start"
              x="6"
            >
              Body
            </text>
          </g>
          <g
            class="node"
            transform="translate(190,384)"
          >
            <circle
              fill="steelblue"
              r="6"
            />
            <text
              dy="0.31em"
              text-anchor="start"
              x="6"
            >
              List
            </text>
          </g>
          <path
            class="link"
            d="M0,256C0,256,95,256,95,256"
            fill="none"
            stroke="green"
            stroke-opacity="0.4"
            stroke-width="1.5"
          />
          <path
            class="link"
            d="M95,256C95,192,190,192,190,128"
            fill="none"
            stroke="green"
            stroke-opacity="0.4"
            stroke-width="1.5"
          />
          <path
            class="link"
            d="M95,256C95,320,190,320,190,384"
            fill="none"
            stroke="green"
            stroke-opacity="0.4"
            stroke-width="1.5"
          />
        </g>
      </svg>
    </div>
  </div>
</DocumentFragment>
`;

exports[`renders & matches snapshot - no props 1`] = `
<DocumentFragment>
  <div
    class="graph-container"
  >
    <h1
      class="h1-absolute"
    >
      Component Tree
    </h1>
    <button
      class="graph-button-1"
    >
      Change View
    </button>
    <div
      id="canvas"
    />
  </div>
</DocumentFragment>
`;
